app/code/VConnect/Blog/Block/PostsList.php

```diff
@@ -48,10 +48,7 @@ public function getPosts(): array
      */
     public function getPostUrl(Post $post): string
     {
-        return $this->urlBuilder->getUrl(
-            'vconnect_blog/post/view',
-            ['id' => $post->getData('entity_id'), '_secure' => true]
-        );
+        return $this->urlBuilder->getBaseUrl() . 'vconnect_blog/' . $post->getUrlKey();
```

Comment:
what if url key would be absent for some reason? how to get a post url this way?